Model/OrderData.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Metrilo\Analytics\Model;
+
+class OrderData
+{
+    public function __construct(
+        \Magento\Sales\Model\ResourceModel\Order\CollectionFactory $orderCollection,
+        \Magento\Directory\Api\CountryInformationAcquirerInterface $countryInterface
+    ) {
+        $this->orderCollection = $orderCollection;
+        $this->countryInterface = $countryInterface;
+    }
+
+    public function getOrders($storeId)
+    {
+        $ordersArray = [];
+        $orders = $this->getOrderQuery($storeId);
+
+        foreach ($orders as $order) {
+            if(!$order->getCustomerEmail()) {
+                continue;
+            }
+
+            $orderItems = $order->getAllItems(); // getAllVisibleItems() returns only parent products (for configurables) from order BUT ignoring child product quantities for BUNDLE products
+
+            foreach ($orderItems as $orderItem) {
+                $itemType = $orderItem->getProductType();
+                if ($itemType != 'configurable' && $itemType != 'bundle') { // exclude configurable/bundle parent product returned by getAllItems() method
```

Comment:
**For the last time** - stop wrapping code in big `if` blocks!

Reply:
Noted. Don't understand why this is such a blind spot.